### PR TITLE
add --inline-client command option

### PR DESCRIPTION
**Pull request by elclanrs**
This is a fix for issue #3361. This PR adds a new `--inline-client` option to customize Webpack's inline client URL that makes live reloading work. This is useful when your are using Vagrant or Docker and the client and host names and/or ports may differ. For example, inside Vagrant I may run `ng serve` on the default 4200 port and reverse proxy to port 80 which I expose in my host as port 1234. But angular-cli prints the wrong URL:

```
http://localhost:4200/sockjs-node/info?t=1480728685812
```

Producing this error:

```
[WDS] Disconnected!
GET http://localhost:4200/sockjs-node/info?t=1480728685812 net::ERR_CONNECTION_REFUSED
```

With this new option I can change it:

```
$ ng serve --inline-client http://localhost:1234
```

The above will use the right URL:

```
http://localhost:1234/sockjs-node/info?t=1480728685812
```

### CODE REVIEW

**Comment by hansl** [comment]
Also please rebase your code with the latest master.

**Comment by tsm91** [comment]
@elclanrs thx, this fixed my issue now i can do `ng serve --inline-client https://projname.tsm.dev`

this should be merged asap, until then i use npm link and keep merging in the latest master

File: packages/angular-cli/commands/serve.ts [hunk 1]
```diff
@@ -103,6 +104,12 @@ const ServeCommand = Command.extend({
       default: false,
       description: 'Enable hot module replacement',
     },
+    {
+      name: 'inline-client',
+      type: String,
+      aliases: ['ic'],
+      description: '(Defaults to "http://host:port")'
```
Comment (tsm91) [inline packages/angular-cli/commands/serve.ts]: `Websocket client URL for Webpack inline mode (Defaults to "http://host:port")` sounds reasonable

**Comment by tsm91** [comment]
@hansl Hi, is there any reason why this hasnt been merged?

I would happily rebase it with the latest master, if you say u can merge it. If you can't merge it, can you please tell why?

**Comment by hansl** [comment]
Looks good, just need to rebase.

**Comment by filipesilva** [comment]
@hansl as per https://github.com/angular/angular-cli/pull/3873#issuecomment-272669659, https://github.com/angular/angular-cli/pull/3952 appears to supersede this PR in a more elegant way, by fixing options that we already have but are not working currently.

**Comment by filipesilva** [comment]
Closed in favor of https://github.com/angular/angular-cli/pull/3952

**Pull request closed without merging**